### ISSUE TITLE
CAD File Sharing System and File Naming Convention

### ISSUE THREAD

**Issue by KyleHelvie** [m2i/MAVRIC-Mechanical#4]
We're working on finding the best system for CAD file sharing. Currently, I am investigating the possibility of using university servers for this. This would mean everyone would need to set up server access on their computers (relatively easy), and all CAD models would have to be created and modified within that folder. Other suggested solutions are appreciated.

In conjunction with this, I am working on a folder organizational system and file naming convention that will keep our CAD models organized.  This is the tentative organizational scheme:

- A "Table of Contents" which functions much like the Folder Structure section in the Github team readmes. It will detail the folder organization so files are easy to find at a glance. Individual CAD files will not be listed on this table of contents.
**The  Folders**
-  Mechanical Team 2017/2018
   - Meeting Documents
      - 9/16/17
         - This Week in Mechanical Team (PDF)
         - Other documents (e.g. if you have a relevant CAD then include it here!)
      - All other meetings
   - Phoenix Rover Design 2017/2018
     - Assembly 
       - Pheonix Assembly 1.0
         - CAD Files
     - Chassis
     - Suspension
        - Suspension Assembly 1.0
        - Suspension Assembly 1.1
     - Electrical Box
     - Robotic Arm
     - Mounting
     - Accessories and Misc.

**The Naming Convention**

_Part Files_
Y#-SSP#L
Y# = Year (e.g. 17)
- Current year
SS = Subassembly Identification Number
- Chassis: 01
- Suspension: 02
- Electrical Box: 03
- Robotic Arm: 04
- Mounting: 05
- Accessories and Misc: 06
P# = Part Number
- The part number should correspond to any drawing files and excel documents.
L = Version (A through Z)

For example, the sheet metal chassis part could be named "17-0101C." This would mean the part was modeled in 2017, it is part of the suspension, it is part #01 and this is the third version of the part. Alternatively, a bushing on the suspension could be named "17-0232A"

*Folders*
There are three types of folders: master, and uploads. Master folders are strictly for organizational purposes. For example, "Pheonix Rover Design 2017/2018" is a master folder as it is meant to hold all design files for Pheonix Rover.  "Suspension Subassembly 1.1" is an upload because it is a folder somebody would upload containing a modified suspension assembly within the 1 series.  Uploads represent individual contributions to the project.

My current plant for folder naming is as follows
- Every subassembly version should have a series number (1 through infinity) and a subseries number. For example, 1.0 would refer to the very first subassembly in the first series. 1.1 would refer to the second subassembly in the first series. A series change represents a substantial change in design, while subseries are modifications.
- Series are entirely separate from part versions. Since CAD parts are left in series folders, there can be as many versions of a particular part within a series to allow for iterative design (e.g. running through different FEA simulations to find optimal design).

 I'm also trying to find a way to associate word/excel documents with parts and subassemblies. This would mean subassemblies and major parts would each have a report explaining basic design decisions. Excel documents would be parts lists containing cost, link (if part purchased), mass estimations, etc.

I am making this an issue because I would love feedback and suggestions. This is a system we'll all have to be using, so I want to make sure it is a system that everybody has a voice in creating.

**Comment by KaDavis5** [comment]
Have we thought about Google drive?

**Comment by rproche** [comment]
I agree that having a top-level contents folder is necessary.  I like the idea of putting relevant CADs to the weekly meetings with the meeting notes.

I feel that the numbering system may be too confusing, though, even if we have a top-level folder explaining it.  We could just have parts named something like
year_month_subassembly_simplepartname_series#.revision#
For example:
2017_sept_suspension_topbushing_2.3
This is much more intuitive to read, even for non-mechanical team members.  I think month is required to better pinpoint the time of drawing, especially between school years.  Also, by naming the parts rather than numbering them, we don't have to mess with assembly files to get the numeric order to match.  Having a series#.revision# will provide more information than just a revision letter.

**Comment by KyleHelvie** [comment]
@KaDavis5 Drive would break assembly relationships and doesn't provide substantial benefits over CyBox.

@rproche The idea is that you can look at the part name at a glance and, once you're used to the code, understand it. I agree that adding an indicator of the month might be valuable. Including a simple part name is included in the wiki entry for this particular naming convention, but I neglected to include it in this comment. I will update the issue to reflect that. What exactly do you mean by "getting the numeric order to match"?

**Comment by rproche** [comment]
@KyleHelvie In assembly drawings, each part has a number.  It isn't a huge concern.  The larger issue is that you would need a table of contents for each sub-assembly to tell people what each part number in the folder means.  Plus, the part numbering is redundant if you are also putting a simple part name in the file name.

**Comment by Talonj123** [comment]
GitHub LFS is configured for up to 1 GB per repository. If your CAD files will not be exceeding this, I would recommend it.

See https://git-lfs.github.com/ for setup.

**Comment by Talonj123** [comment]
Essentially, you would set your CAD files (.stl I think, maybe something else) to use LFS and then do everything else as normal. Let me know if you want to do this and I can do the setup.

**Comment by dwhited** [comment]
Uploading to Github would be useful for sharing concepts across the team, but once you upload it, it would be best to consider that revision locked. Once a model is removed from a local or network copy (uploading to Github or CyBox), it's not really possible to edit them or pull them back to your computer like word or excel files. Plus then you break assembly relations by doing that, or open yourself up to corrupting the entire rover model. The best case scenario is getting control of the COE server folder, because everything can be saved to that rather than using a local copy, and you can access that from any computer that uses Solidworks. We can set up access using the ISU ASW lists, we would just need to work with Engineering IT. That way Matt can have control of it and add people when we ask.

**Comment by GageLochner** [comment]
I would suggest only uploading the model to Github once it has begun the manufacturing process. That way, we have the model for the part that ends up being made on record.

I highly encourage the folder and naming system. It should make things easier. 

I would also encourage that you keep a copy of any documentation (excel or word docs) on github. This way, the documentation sticks with the model for the manufactured part. 

I would love to see a write up on how and why each manufactured part is the way it is, but this might be a stretch considering how much of the engineering work is already done for this semester.

**Comment by KyleHelvie** [comment]
@GageLochner 
Documentation is being kept track of on the fileserv, and is associated with each assembly (part in some cases). I also hoped to have the "why" documentation, but I've yet to formulate a formal way of doing this. This might be something to ask our Boeing representatives about. I'll see about having a date where I periodically update the github from the fileserv.
